src/data/move.ts

```diff
@@ -5359,7 +5361,8 @@ export class SuppressAbilitiesIfActedAttr extends MoveEffectAttr {
 export class TransformAttr extends MoveEffectAttr {
   apply(user: Pokemon, target: Pokemon, move: Move, args: any[]): Promise<boolean> {
     return new Promise(resolve => {
-      if (!super.apply(user, target, move, args)) {
+      if (!super.apply(user, target, move, args) || target.illusion.active || user.illusion.active) {
+        user.scene.queueMessage("But it failed!");
```

Comment:
```suggestion
        user.scene.queueMessage(i18next.t("battle:attackFailed"));
```
Need to localize and code already exists.